components/TemplateGrid/TemplateGridContainer.vue

```diff
@@ -45,202 +45,142 @@ export default {
         },
     },
     data: () => ({
-        ghostIndex: 'ghost_item',
         positionBeetwenRows: 0.5,
-        lastColumn: null,
-        lastRow: null,
+        ghostElement: {
+            id: 'ghost_item',
+            column: null,
+            row: null,
+            ghost: true,
+        },
+        mousePosition: {
+            overColumn: null,
+            overRow: null,
+            collisionRelation: null,
+            shadowItem: null,
+        },
+        debounceFunc: null,
     }),
     computed: {
         dataWithoutGhostElement() {
-            return this.gridData.filter(element => element.id !== this.ghostIndex);
-        },
-        ghostElement() {
-            // Last row might be 0 - that's why === null
-            return this.lastRow === null ? null : this.gridData[this.lastRow];
+            return this.gridData.filter(element => element.id !== this.ghostElement.id);
         },
         maxRow() {
             return Math.max(...this.dataWithoutGhostElement.map(item => item.row));
```

Comment:
I would suggest you to use getMaxObjectValueInArrayByObjectKey from arrayWrapper